dev-support/git-hooks/post-checkout/100-update-submodules.sh

```diff
@@ -24,35 +25,17 @@ set -o nounset
 
 bin="$(cd "$(dirname "$0")" > /dev/null; pwd)"
 
-accord_repo='https://github.com/apache/cassandra-accord.git'
-accord_branch='1230eceb077c928123e6bf848a103964fe90c9f7'
-accord_src="$bin/cassandra-accord"
-
-checkout() {
-  cd "$accord_src"
-    git checkout "$accord_branch"
-    echo "$accord_branch" > .BRANCH
-  cd -
-}
-
 _main() {
-  # have we already cloned?
-  if [[ ! -e "$accord_src" ]] || [[ $(cat "$accord_src/.REPO" || true) != "$accord_repo" ]]; then
-    rm -rf "$accord_src" || true
-    git clone "$accord_repo" "$accord_src"
-    echo "$accord_repo" > "$accord_src/.REPO"
-    checkout
-  fi
-  if [[ $(cat "$accord_src"/.BRANCH || true) != "$accord_branch" ]]; then
-    checkout
-  fi
-  cd "$accord_src"
-  # are there changes?
-  git pull --rebase origin "$accord_branch"
-  if [[ $(git rev-parse HEAD) != $(cat .SHA || true) ]]; then
-    ./gradlew clean install -x test
-    git rev-parse HEAD > .SHA
+  # In case the usage happens at a different layer, make sure to cd to the toplevel
+  local root_dir
+  root_dir="$(git rev-parse --show-toplevel)"
+  cd "$root_dir"
+
+  if [[ ! -e .gitmodules ]]; then
+    # nothing to see here, look away!
+    return 0
   fi
+  git submodule update --init --recursive
```

Comment:
originally didn't have `--init` but this is needed if someone adds a new submodule... `--init` becomes a noop if the submodule is already init, so safe to include.

We don't need `--recursive` right now, but since the hooks no-op if they exist figured if we ever go that route best to add now...

Reply:
this is failing for me, on my initial switch to the branch i'm getting.
```
❯ git sw accord-submodules
Updating files: 100% (3001/3001), done.
Switched to branch 'accord-submodules'
Your branch is up to date with 'dcapwell/accord-submodules'.
Cloning into '/Users/mick/src/apache/cassandra/accord'...
fatal: transport 'file' not allowed
fatal: Fetched in submodule path 'accord', but it did not contain fa73c60fd132b817fe594e8e42bdd1329628b626. Direct fetching of that commit failed.
```

if i first do
```
❯ git config --global protocol.file.allow always
```
then the first fatal message is removed, but it still does not work…

Reply:
just me. fetching first the `dcapwell` remote fixed it. (odd error msgs though)

Reply:
we spoke in slack, I sadly can't repo and @michaelsembwever and I have the same `git` version... tested different files named `accord` and didn't work..

I have 1 theory based off more testing, if C* commits an accord sha but that sha was never pushed up, then you can hit the last line....  to help prevent the cases I detected I added git hooks to make sure your SHA is pushed before you commit.